saltapi/repository/user_repository.py

```diff
@@ -735,6 +737,8 @@ def find_user_with_username_and_password(
             raise NotFoundError()
         if not self.verify_password(password, user.password_hash):
             raise NotFoundError()
+        if not self.user_verified(user.id):
```

Comment:
Raising an error is an option, but I don't think it should be a ValidationError.

You should also raise an error in case the user is inactive.